### DISCUSSION TITLE
To make server with expressjs, apollo-server, and graphql

### DISCUSSION THREAD

**Discussion by naeemkhan9293** [apollographql/apollo-server#7797] [General]
const express = require("express");
const { createHandler } = require("graphql-http/lib/use/express");
const cors = require("cors");
const schema = require("./schema");
const { ApolloServer } = require("@apollo/server");
const { expressMiddleware } = require("@apollo/server/express4");
const http = require("http");
const app = express();
const {
  ApolloServerPluginDrainHttpServer,
} = require("@apollo/server/plugin/drainHttpServer");
const port = 4000;
const httpServer = http.createServer(app);
const bodyParser = require("body-parser");

const corsOptions = {
  origin: "http://localhost:4000/",
  optionsSuccessStatus: 200, // some legacy browsers (IE11, various SmartTVs) choke on 204
};

const server = new ApolloServer({
  schema,
  plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
});

server.start().then(() => {
  app.use(cors(), bodyParser.json(), expressMiddleware(server));
  httpServer.listen(port, () => {
    console.log(`🚀 Server ready at http://localhost:${port}`);
  });
});

**Comment by qu8n** [comment]
@naeemkhan9293 I recommend using the [apollo-server-express](https://www.apollographql.com/docs/apollo-server/migration/#migrate-from-apollo-server-express) package instead to initialize your Apollo Sever. It was made specifically for the Apollo Server to work within an Express application.

Example sourced from the docs:
```
// npm install @apollo/server express graphql cors
import { ApolloServer } from '@apollo/server';
import { expressMiddleware } from '@apollo/server/express4';
import { ApolloServerPluginDrainHttpServer } from '@apollo/server/plugin/drainHttpServer';
import express from 'express';
import http from 'http';
import cors from 'cors';
import { typeDefs, resolvers } from './schema';

const app = express();
const httpServer = http.createServer(app);
const server = new ApolloServer({
  typeDefs,
  resolvers,
  plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
});
await server.start();
app.use(
  '/graphql',
  cors(),
  express.json(),
  expressMiddleware(server, {
    context: async ({ req }) => ({ token: req.headers.token }),
  }),
);

await new Promise((resolve) => httpServer.listen({ port: 4000 }, resolve));
console.log(`🚀 Server ready at http://localhost:4000/graphql`);
```

**Comment by trevor-scheer** [comment]
@naeemkhan9293 what is the question? Is this answered by @qu8n's example?
